### PR TITLE
Reduce code duplication in Scanner

### PR DESCRIPTION
**Pull request by rh-id**
simplifying duplicate HostProcess for readability

### CODE REVIEW

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Scanner.java [hunk 1]
```diff
@@ -325,6 +294,21 @@ public void scan(Target target) {
         }
     }
 
+	private HostProcess getAndInitHostProcess(String hostAndPort, StructuralNode node) {
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/Scanner.java:297]: `createHostProcess`

Needs a ZAP comment after the license header.

Comment (rh-id) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/Scanner.java:297]: Hi, so I need to add `// ZAP: 2020/05/19 Remove HostProcess code duplication` after the `// ZAP: 2019/11/09 Ability to filter to active scan (Issue 5278)` ?

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/Scanner.java]: That works.

**Comment by thc202** [comment]
The building is failing because of the code format (you'll have to run `gradlew spotlessApply` to address it).

**Comment by thc202** [comment]
Thanks!

**Pull request merged**